x-pack/plugins/infra/public/hooks/use_trial_state.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useEffect } from 'react';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+import { API_BASE_PATH as LICENSE_MANAGEMENT_API_BASE_PATH } from '../../../license_management/common/constants';
+
+export enum TrialStatusLoadState {
+  Loading = 'loading',
+  Ok = 'ok',
+  Error = 'error',
+}
+
+interface UseTrialStatusState {
+  loadState: TrialStatusLoadState;
+  isTrialAvailable: boolean;
+}
+
+export function useTrialStatus(): UseTrialStatusState {
+  const { services } = useKibana();
+
+  const [loadState, setLoadState] = useState<TrialStatusLoadState>(TrialStatusLoadState.Loading);
+  const [isTrialAvailable, setIsTrialAvailable] = useState<boolean>(false);
+
+  useEffect(() => {
```

Comment:
It's my understanding that we want to move away from "self initiating" `useEffect` calls, as it makes it harder to debug things down the line. Logs source uses an `initialize` call instead: https://github.com/elastic/kibana/blob/master/x-pack/plugins/infra/public/containers/logs/log_source/log_source.ts#L142. Or, if we do keep the `useEffect` could we move it to the consumer, so it's more explicit, rather than within the data fetching hook?